doc/source/conf.py

```diff
@@ -416,3 +418,55 @@ def linkcode_resolve(domain, info):
 SphinxDocString._str_examples = _rng_html_rewrite(
     SphinxDocString._str_examples
 )
+class LegacyDirective(Directive):
+    """
+    Adapted from docutils/parsers/rst/directives/admonitions.py
+
+    Uses a default text if the directive does not have contents. If it does,
+    the default text is concatenated to the contents.
+
+    """
+    has_content = True
+    node_class = nodes.admonition
+    optional_arguments = 1
+
+    def run(self):
+        try:
+            obj = self.arguments[0]
+        except IndexError:
+            # Argument is empty; use default text
+            obj = "submodule"
+        text = (f"This {obj} is now considered legacy and will no longer "
+                "receive updates.")
```

Comment:
Should we clarify that, not receiving updates could mean a removal at some point?

In a follow up we should have a section about legacy and deprecation clarifying our policy. (Could be alongside the existing section we have in the core dev part of the contributing guide.)